File: WeexBox/Module/Event/EventModule.m
//
//  EventModule.m
//  WeexBox
//
//  Created by Mario on 2018/8/18.
//  Copyright © 2018年 Ayg. All rights reserved.
//

#import "EventModule.h"

#pragma clang diagnostic ignored "-Wundeclared-selector"
@implementation EventModuleOC

WX_EXPORT_METHOD(@selector(register:callback:))
WX_EXPORT_METHOD(@selector(emit:))
WX_EXPORT_METHOD(@selector(unregister:))
WX_EXPORT_METHOD(@selector(unregisterAll:))

@end
#pragma clang diagnostic pop

File: WeexBox/Module/Event/EventModule.swift
//
//  EventModule.swift
//  WeexBox
//
//  Created by Mario on 2018/8/18.
//  Copyright © 2018年 Ayg. All rights reserved.
//

import Foundation

class EventModule: EventModuleOC {
    
    @objc func register(_ name: Any, callback: WXModuleKeepAliveCallback?) {
        Event.register(target: getVC(), name: WXConvert.nsString(name)) { notification in
            callback?(notification?.userInfo, true)
        }
    }
    
    @objc func emit(_ params: Dictionary<String, Any>) {
        let options = JsOptions.deserialize(from: params)!
        Event.emit(name: options.name!, info: options.params)
    }
    
    @objc func unregister(_ name: Any) {
        Event.unregister(target: self, name: WXConvert.nsString(name))
    }
    
    @objc func unregisterAll() {
        Event.unregisterAll(target: self)
    }
    
}
